dyndns: Add LuaDNS provider

aaron-sierra opened the pull request:
- [x] Redmine Issue: https://redmine.pfsense.org/issues/15089
- [x] Ready for review

Complement ACME support for LuaDNS by adding Dynamic DNS support for the provider.

If the updates refactoring host checking for other providers (Cloudflare, Gandi LiveDNS, Linode, Porkbun, and Yandex) are, too much to consider, I can easily drop those three commits without issue.

Apex, wildcard, and simple hosts have been tested.
IPv4 and IPv6 support have been tested.

aaron-sierra commented:
I'm currently locked out of my Redmine account, but I'll update the PR when I get in and create the ticket.

aaron-sierra commented:
> I'm currently locked out of my Redmine account, but I'll update the PR when I get in and create the ticket.

For the record, I realized my Redmine account mistake, created a Redmine ticket, and updated this pull request description with the link on December 12, 2023.

liberodark commented:
Hi,

Why any review of this PR is make after 7 mouths ?
This is not a criticism, I would just like to understand why this addition is not effective if the code is correct and error-free?
Thank you in advance if anyone takes the time to respond to my message.

Best Regards

marcos-ng commented:
Thank you for the contribution. This will need to be refactored given the recent changes to the DDNS class.

Also, keep the this MR specific to adding the feature. Improvements and other changes to the DDNS class in general should be left to a separate redmine/MR.

aaron-sierra commented:
> This will need to be refactored given the recent changes to the DDNS class.

I've accounted for the minor changes since last year and updated the "Last Tested" note. I've also dropped changes not directly related to LuaDNS support.

marcos-ng commented:
Thank you! Now that the rest of the work for DDNS is done, this should be good to go in after the following changes:

* Rebase the branch.
* change `$recordType = $this->_useIPv6 ? "AAAA" : "A";` to `$recordType = ($this->_addressFamilyRR == AF_INET6) ? "AAAA" : "A";`.
* There are references to `$this->_logPrefix` which can simply be removed.
* Make sure to squash commits and include the redmine ID in the message, e.g.: `dyndns: Add LuaDNS provider. Implement #15089`.

aaron-sierra commented:
* [ ] Rebase
* [ ] Account for `$this->_addressFamilyRR`
* [ ] Drop references to `$this->_logPrefix`
* [ ] Update "Last Tested"
* [ ] Add linkage to Redmine ticket
